### ISSUE TITLE
[C#] Implement new Concept Exercise: regular-expressions

### ISSUE THREAD

**Issue by mikedamay** [exercism/v3#1638]
This issue describes how to implement the `regular-expressions` concept exercise for the C# track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student how to use regular expressions in C#.

## Learning objectives

- Know how to use regular expressions with `RegEx` in C#.

## Out of scope

- Intermediate and advanced regular expression patterns.

## Concepts

- `regular-expressions` as they are handled in C# and .NET.

## Prerequisites

- `strings`

Other prerequisites will depend on the story.

### Hints
[regular expressions][regular-expressions] documentation describes regexes and the flavour built into the .NET libraries.
[Regex][regex] documentation describing the built-in library class.

### After

[regular expressions][regular-expressions] documentation describes regexes and the flavour built into the .NET libraries.
[Regex][regex] documentation describing the built-in library class.

## Representer

This exercise does not require any specific representation logic to be added to the [representer][representer].

## Analyzer

This exercise does not require any specific analyzer logic to be added to the [analyzer][analyzer].

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/csharp/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[regular-expressions]: https://docs.microsoft.com/en-us/dotnet/standard/base-types/regular-expression-language-quick-reference
[regex]: https://docs.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.regex?view=netcore-3.1
[analyzer]: https://github.com/exercism/csharp-analyzer
[representer]: https://github.com/exercism/csharp-representer

**Comment by mikedamay** [comment]
I'm thinking: analyse the logs produced in the `strings` or some variant of them.

Comments welcome.

**Comment by mikedamay** [comment]
closed by PR #1858
